x-pack/plugins/apm/server/routes/service_groups/get_services_counts.ts

```diff
@@ -7,49 +7,67 @@
 
 import { ProcessorEvent } from '@kbn/observability-plugin/common';
 import { rangeQuery, kqlQuery } from '@kbn/observability-plugin/server';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
 import { Setup } from '../../lib/helpers/setup_request';
 import { SERVICE_NAME } from '../../../common/elasticsearch_fieldnames';
+import { SavedServiceGroup } from '../../../common/service_groups';
 
 export async function getServicesCounts({
   setup,
-  kuery,
-  maxNumberOfServices,
   start,
   end,
+  serviceGroups,
 }: {
   setup: Setup;
-  kuery: string;
-  maxNumberOfServices: number;
   start: number;
   end: number;
+  serviceGroups: SavedServiceGroup[];
 }) {
   const { apmEventClient } = setup;
 
-  const response = await apmEventClient.search('get_services_count', {
+  const serviceGroupsKueryMap: Record<string, QueryDslQueryContainer> =
+    serviceGroups.reduce((acc, sg) => {
+      return {
+        ...acc,
+        [sg.id]: kqlQuery(sg.kuery)[0],
+      };
+    }, {});
+
+  const params = {
     apm: {
-      events: [
-        ProcessorEvent.metric,
-        ProcessorEvent.transaction,
-        ProcessorEvent.span,
-        ProcessorEvent.error,
-      ],
+      events: [ProcessorEvent.metric],
```

Comment:
I'm not necessarily against querying only the metric data, but it does come with the caveat that services that only have errors are not included. It's an edge case though, they'd have to actively disable system/app metrics too. Can we leave a comment though here to explain that reasoning?